### PR TITLE
feat: update camp model to have on object for location

### PR DESCRIPTION
**Pull request by uyousafzai54**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Camp-model-update-f04d52deb7774f828c544c660b03a63b)

## Implementation description
```
{
	streetAddress1: string
	streetAddress2: string (optional)
	city: string
	province: string
	postalCode: string
}
```
* Updated the location model to the new schema shown above
* Updated the CampDTO 
* Updated the Camp Validator
* Updated automated tests for creating and updating camps 

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/campValidators.ts [hunk 1]
```diff
@@ -70,8 +70,32 @@ export const createCampDtoValidator = async (
   if (!validatePrimitive(body.active, "boolean")) {
     return res.status(400).send(getApiValidationError("active", "boolean"));
   }
-  if (!validatePrimitive(body.location, "string")) {
-    return res.status(400).send(getApiValidationError("location", "string"));
+  if (body.location) {
+    if (!validatePrimitive(body.locaton.streetAddress1, "string")) {
```
Comment (annechung) [inline backend/typescript/middlewares/validators/campValidators.ts:74]: > locaton
bruh

File: backend/typescript/models/camp.model.ts [hunk 2]
```diff
@@ -18,7 +18,13 @@ export interface Camp extends Document {
   fileName?: string;
   formQuestions: (Schema.Types.ObjectId | FormQuestion)[];
   latePickup: string;
-  location: string;
+  location: {
+    streetAddress1: string;
+    streetAddress2: string;
+    city: string;
+    province: string;
+    postalCode: string;
```
Comment (annechung) [inline backend/typescript/models/camp.model.ts:26]: can you set everything to required except for streetAddress2

File: backend/typescript/middlewares/validators/campValidators.ts (hunk 1 above)
Comment (uyousafzai54) [reply backend/typescript/middlewares/validators/campValidators.ts:74]: bruh

Comment (NirmalHegde) [reply backend/typescript/middlewares/validators/campValidators.ts]: bruh

**Review by annechung — CHANGES_REQUESTED** [review]
good start! you should update the camp creation/edit methods to also pass in these new address fields

**Review by annechung — APPROVED** [review]
nice work :)

**Pull request merged**